File: src/main/java/com/awsblog/queueing/model/DequeueResult.java
package com.awsblog.queueing.model;

import com.awsblog.queueing.appdata.Assignment;
import com.awsblog.queueing.utils.Utils;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Result for the peek() API call
 * 
 * @author zorani
 *
 */
@JsonInclude(Include.NON_NULL)
public class DequeueResult extends ReturnResult {

	/**
	 * Default empty c-tor
	 */
	public DequeueResult() {
		
		super();
	}

	/**
	 * C-tor
	 * 
	 * @param id
	 */
	public DequeueResult(String id) {
		
		super(id);
	}
	
	/**
	 * Get the values from ReturnResult object
	 * 
	 * @param result
	 * @return
	 */
	public static DequeueResult fromReturnResult(ReturnResult result) {
		
		Utils.throwIfNullObject(result, "Resulting object is NULL!");
		
		DequeueResult dequeueResult = new DequeueResult(result.getId());
		dequeueResult.setLastUpdatedTimestamp(result.getLastUpdatedTimestamp());
		//dequeueResult.setStatus(result.getStatus());
		dequeueResult.setVersion(result.getVersion());
		dequeueResult.setReturnValue(result.getReturnValue());
		
		return dequeueResult;
	}
		
	/**
	 * @return the dequeuedAssignmentObject
	 */
	public Assignment getDequeuedAssignmentObject() {
		return dequeuedAssignmentObject;
	}

	/**
	 * @param dequeuedAssignmentObject the dequeuedAssignmentObject to set
	 */
	public void setDequeuedAssignmentObject(Assignment dequeuedAssignmentObject) {
		this.dequeuedAssignmentObject = dequeuedAssignmentObject;
	}

	/**
	 * Get the peeked Assignment id
	 * 
	 * @return
	 */
	@JsonProperty("dequeue_id")
	public String getPeekedAssignmentId() {
		
		if (Utils.checkIfNotNullObject(this.dequeuedAssignmentObject))
			return this.dequeuedAssignmentObject.getId();
		
		return "NOT FOUND";
	}

	// ---------------- fields

	@JsonIgnore
	private Assignment dequeuedAssignmentObject = null;
	
} // end DequeueResult

File: src/main/java/com/awsblog/queueing/model/DequeueResult.kt
package com.awsblog.queueing.model

import com.awsblog.queueing.appdata.Assignment
import com.awsblog.queueing.utils.Utils
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Result for the peek() API call
 *
 * @author zorani
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
class DequeueResult : ReturnResult {
    /**
     * Default empty c-tor
     */
    constructor() : super() {}

    /**
     * C-tor
     *
     * @param id
     */
    constructor(id: String?) : super(id) {}

    /**
     * Get the peeked Assignment id
     *
     * @return
     */
    @get:JsonProperty("dequeue_id")
    val peekedAssignmentId: String?
        get() = if (Utils.checkIfNotNullObject(dequeuedAssignmentObject)) dequeuedAssignmentObject!!.id else "NOT FOUND"
    /**
     * @return the dequeuedAssignmentObject
     */
    /**
     * @param dequeuedAssignmentObject the dequeuedAssignmentObject to set
     */
    // ---------------- fields
    @JsonIgnore
    var dequeuedAssignmentObject: Assignment? = null

    companion object {
        /**
         * Get the values from ReturnResult object
         *
         * @param result
         * @return
         */
        fun fromReturnResult(result: ReturnResult): DequeueResult {
            Utils.throwIfNullObject(result, "Resulting object is NULL!")
            val dequeueResult = DequeueResult(result.id)
            dequeueResult.lastUpdatedTimestamp = result.lastUpdatedTimestamp
            //dequeueResult.setStatus(result.getStatus());
            dequeueResult.version = result.version
            dequeueResult.returnValue = result.returnValue
            return dequeueResult
        }
    }
} // end DequeueResult
